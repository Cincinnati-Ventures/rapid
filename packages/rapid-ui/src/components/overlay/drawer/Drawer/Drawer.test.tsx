import React from 'react';
import { render } from '@testing-library/react';
import { expect, it, describe } from 'vitest';
import {
	Drawer,
	DrawerContent,
	DrawerHeader,
	DrawerBody,
	DrawerFooter,
	DrawerCloseButton,
} from '../';

describe('Modal', () => {
	it('should render default correctly', () => {
		const { container } = render(
			<Drawer
				open
				direction='right'
				onClose={() => {
					console.log('closing drawer');
				}}
				size='md'
			>
				<DrawerContent>
					<DrawerHeader>
						<text>Drawer Title</text>
						<DrawerCloseButton />
					</DrawerHeader>
					<DrawerBody>
						<text>
							This is the drawer content. You can add any elements
							here to display inside the drawer.
						</text>
					</DrawerBody>
					<DrawerFooter>
						<button>Close Drawer</button>
					</DrawerFooter>
				</DrawerContent>
			</Drawer>,
		);
		expect(container).toMatchSnapshot();
	});

	it('should render correctly with non default props', () => {
		const { container } = render(
			<Drawer styles='bg-green' open onClose={() => {}} />,
		);
		expect(container).toMatchSnapshot();
	});
});
