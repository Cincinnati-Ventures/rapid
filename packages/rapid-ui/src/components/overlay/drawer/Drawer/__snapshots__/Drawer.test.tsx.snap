// Vitest Snapshot v1

exports[`Modal > should render correctly with non default props 1`] = `<div />`;

exports[`Modal > should render default correctly 1`] = `<div />`;
